feat(pattern): created an alpha version of the cards with images patter #705

mkothur opened the pull request:
### Related Ticket(s)

Related to issue #705
### Description

Created the alpha version of the Cards with images and made changes in `/packages/react/src/components/CardLink` added one pore property `source` and function to render an image in `CardsWithImageGroup`

![image](https://user-images.githubusercontent.com/53789187/70557505-ee71f200-1b50-11ea-804d-4746f7a6a450.png)
![image](https://user-images.githubusercontent.com/53789187/71521604-782ef880-288f-11ea-9ac4-4bad2db5fe84.png)
![image](https://user-images.githubusercontent.com/53789187/71521619-841aba80-288f-11ea-9da9-44753b9fd21c.png)


### Changelog

**Changed**

- Added CardWithImage component and styles updated the CardLink react componet.
- Added a new feature flag for the created component.

jeffchew commented on packages/patterns-react/src/internal/FeatureFlags.js:
```diff
@@ -77,3 +77,11 @@ export const DDS_CARDS_WITHOUT_IMAGES =
+ * @type {string | boolean}
+ */
+export const DDS_CARDS_WITH_IMAGES =
+  process.env.CARD === 'true' || DDS_FLAGS_ALL || false;
```
@mkothur should be `process.env.DDS_CARDS_WITH_IMAGES`

mkothur replied:
Yeah I will update

annawen1 replied:
Can you also add `DDS_CARDS_WITH_IMAGES` to the `.env.example` file within the patterns-react package?

annawen1 commented on packages/styles/scss/patterns/cards-with-images/_cards-with-images-group.scss:
```diff
@@ -0,0 +1,116 @@
+  .bx--card-link {
+    background-color: $ui-01;
+    border-color: $ui-02;
+    outline: 1px solid $ui-background;
```
can you overwrite the padding from the card link component by adding:
` padding: 0;` here?

mkothur replied:
Okay.

jeffchew commented on packages/patterns-react/src/patterns/CardWithImages/CardsWithImages.js:
```diff
@@ -0,0 +1,129 @@
+        el.style.height = maxHeight + 'px';
+      });
+    });
+  };
```
@RaphaelAmadeu @sannminwin @kennylam @annawen1 @emerson-pereira @andysherman2121 @carthick @photodow I feel like this has been created a number of times. Should we move this into `utilities`?

sannminwin replied:
I think getting the hightest div, making all div equal height etc .. are very common scenario in frontend work and we should get this tool handy in `utilities`.

annawen1 replied:
Agreed since it's being used in multiple places

jeffchew replied:
@mkothur can you move this into `packages/utilities`?

photodow replied:
I was just talking to @RaphaelAmadeu about this the other day. Yeah, I 100% agree that this needs to be in a utility. I can create an issue for it: https://github.com/carbon-design-system/ibm-dotcom-library/issues/816

photodow replied:
One other thing to note on this is I'm seeing we're doing a lot of resizing of the entire card itself. This is causing some improper alignment in the card's content area.

![image](https://user-images.githubusercontent.com/3793636/70641953-09ea0500-1c04-11ea-8e52-2cd7cde157ae.png)

If instead, we match the height of all the titles and body text within the card everything should align to what the specs are asking for along with the same height of the cards themselves.

![image](https://user-images.githubusercontent.com/3793636/70642385-bcba6300-1c04-11ea-9202-eb96b41921f8.png)

RaphaelAmadeu replied:
I guess the alignment issue can be solved with flex. If we make the utility this specific it would  be limited by the patterns.

jeffchew commented on packages/patterns-react/src/patterns/CardWithImages/README.md:
```diff
@@ -0,0 +1,79 @@
+import ReactDOM from 'react-dom';
+import { CardsWithImages } from '@carbon/ibmdotcom-patterns-react';
+import 'yourapplication.scss';
+import '@carbon/ibmdotcom-styles/scss/patterns/cards-with-images/index.scss';
```
This should be moved under the `scss` example file instead:

```scss
// yourapplication.scss
@import '@carbon/type/scss/font-face/mono';
@import '@carbon/type/scss/font-face/sans';
@include carbon--font-face-mono();
@include carbon--font-face-sans();

@import '@carbon/ibmdotcom-styles/scss/patterns/cards-with-images/index';
```

jeffchew replied:
@mkothur doesn't look like this has been done yet?

jeffchew commented on packages/react/src/components/CardLink/CardLink.js:
```diff
@@ -36,13 +44,34 @@ const CardLink = ({ title, href, content, icon, className, ...props }) => {
+    <img
+      src={source}
+      className={`${prefix}--card-link__image`}
+      alt="cards with"
```
we should open up a prop for the alt text

jeffchew commented on packages/react/src/components/CardLink/CardLink.js:
```diff
@@ -36,13 +44,34 @@ const CardLink = ({ title, href, content, icon, className, ...props }) => {
+
+  return (
+    <img
+      src={source}
```
`source` seems a bit too vague. Should use something like `imgSrc` and `imgAltText`, but check with @photodow

mkothur replied:
Sure @jeffchew

annawen1 replied:
Any reason why we can't just do 
```
{source && <img src={source} classname={`${prefix}--card-link__image`} alt={alt} />}
```

or is it cleaner to look at with the `renderImage` func?

photodow replied:
@jeffchew I've been wondering if there is a way we could unify the data model to be a little more consistent across the board. Almost everything (up to this point) has a very similar content structure, just some things are optional. For an image (still in progress #767) I think it would be great if we could use something like this.

```js
{
  src: "https://example.com/img.jpg",
  alt: "example image"
  ...
  // anything else needed for the image (e.g. maybe aria attributes, or title?)
  ...
}
```

So it would be a little more like this... `<img src="image.src" alt="image.alt"/>`. Or you could even use a spread here I would think, maybe... `<img {...image} />`

jeffchew replied:
@photodow is the idea to use the responsive image component for these types of things, just have a bunch of optional fields for it?

photodow replied:
Yes, that's what I'm hoping for, so we can keep things consistent.

jeffchew commented on packages/patterns-react/src/patterns/CardWithImages/README.md:
```diff
@@ -0,0 +1,79 @@
+| ----------- | ------------------------------------------------------------------- |
+| `title`     | Cards with images title                                             |
+| `groupCard` | groupCard, array with href, text and target properties              |
+| `cards`     | Cards, array of objects with image, title, copy and link properties |
```
@mkothur seems a bit vague what can be passed exactly, @annawen1 do we have an example that includes the structure of objects that are getting passed in?

Also, if there are updates to `CardLink`, then that README should get updated too.

mkothur replied:
Okay

annawen1 commented on packages/styles/scss/patterns/cards-with-images/_cards-with-images-group.scss:
```diff
@@ -0,0 +1,116 @@
+  .#{$prefix}--cards-with-images-group {
+    padding-bottom: $carbon--layout-05;
+    &__title {
+      @include type-style('expressive-heading-02');
```
Following the sketch file, we need 64px under the title here:
<img width="709" alt="Screen Shot 2019-12-10 at 4 03 52 PM" src="https://user-images.githubusercontent.com/54281166/70570934-d909c180-1b6a-11ea-8e7e-9a243fbdfbb3.png">

Can you add `padding-bottom: $layout-05` here?

mkothur replied:
Okay

annawen1 commented on packages/patterns-react/src/patterns/CardWithImages/__stories__/CardsWithImages.stories.js:
```diff
@@ -0,0 +1,96 @@
+      return (
+        <CardsWithImages
+          cardsGroup={object('cardsGroup', cardsGroup)}
+          theme={select('theme', themes, themes['g10'])}
```
we don't need the brackets here, just `themes.g10`

mkothur replied:
Okay

annawen1 commented on packages/patterns-react/src/patterns/CardWithImages/__stories__/CardsWithImages.stories.js:
```diff
@@ -0,0 +1,96 @@
+
+      const themes = {
+        g10: 'g10',
+        'light (white)': '',
```
Can you also just simplify to just `white` instead of `'light (white)'`

mkothur replied:
Sure

mkothur commented:
@jeffchew, @annawen1: addressed the above comments please do a review. Thanks!

jeffchew commented on packages/react/src/components/CardLink/__stories__/Card.stories.js:
```diff
@@ -4,17 +4,20 @@ import { withKnobs, select, text } from '@storybook/addon-knobs';
-        <div className="bx--row">
-          <div className="bx--col-sm-2 bx--col-md-3 bx--col-lg-6 bx--col-xlg-4 bx--no-gutter">
-            <Story />
+if (DDS_CARD_LINK) {
```
@mkothur we have removed the feature flag for CardLink.

mkothur replied:
Okay, I will revert back to that.

jeffchew commented on packages/react/src/components/CardLink/README.md:
```diff
@@ -47,6 +47,8 @@ ReactDOM.render(<App />, document.querySelector('#app'));
 | `icon`      | NO       | Element   | null          | Provide an optional icon to the footer from [Carbon's icon library](https://www.carbondesignsystem.com/guidelines/icons/library) |
 | `title`     | YES      | String    | n/a           | Concise yet descriptive string of text describing the linked resource.                                                           |
+| `imgSrc`    | NO       | String    | null          | Image source to be passed as a property to the to the CardLink component                                                         |
+| `altText`   | NO       | String    | null          | Image alternate text to be passed as a property to the to the CardLink component                                                 |
```
@photodow just checking if these changes are something you are tracking for the CardLink component?

mkothur replied:
Sure will move _matchHeight into `packages/utilities`

jeffchew commented on packages/styles/scss/components/card-link/index.scss:
```diff
@@ -23,6 +23,10 @@
+    &__image {
+      height: 192px;
+    }
+
```
@mkothur are we fixing height like this?
cc: @photodow

mkothur replied:
sorry, now I changed to cards-with-images-group.scss file

photodow replied:
Trying to define this more in #768. Not sure if this will be defined in this pattern, or not and maybe more based on ratio. We'll see, this should be fine for this for the moment.

jeffchew commented on packages/styles/scss/patterns/cards-with-images/index.scss:
```diff
@@ -0,0 +1,6 @@
+@import './cards-with-images-group';
+
+@include cards-with-images;
+@include cards-with-images-group;
```
Are you also importing this file into `ibm-dotcom-styles.scss`?

mkothur replied:
No. I will import here also

jeffchew commented on packages/patterns-react/src/patterns/CardWithImages/index.js:
```diff
@@ -0,0 +1,8 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+
+export { default as CardsWithImages } from './CardsWithImages';
```
@mkothur you also need to export from `src/patterns/index.js`:

```
export * from './CardWithImages`;
```

Also, shouldn't the folder be called `CardsWithImages` and not `CardWithImages`?

mkothur replied:
Okay

jeffchew commented:
@photodow @sannminwin @mkothur a general question, does Cards With Images need to be a completely separate layout pattern from Cards Without Images?

mkothur commented on packages/patterns-react/src/patterns/CardWithImages/__stories__/CardsWithImages.stories.js:
```diff
@@ -0,0 +1,101 @@
+import { storiesOf } from '@storybook/react';
+import { DDS_CARDS_WITH_IMAGES } from '../../../internal/FeatureFlags';
+import { withKnobs, select, object } from '@storybook/addon-knobs';
+import '../../../../../styles/scss/patterns/cards-with-images/index.scss';
```
Sure

mkothur commented on packages/patterns-react/src/patterns/CardWithImages/README.md:
```diff
@@ -0,0 +1,79 @@
+
+function App() {
+  return;
+  <CardsWithImages cardsGroup={cardsGroup} />;
```
yeah, it is my bad.

mkothur commented:
> @photodow @sannminwin @mkothur a general question, does Cards With Images need to be a completely separate layout pattern from Cards Without Images?

mkothur commented:
@jeffchew: yeah, I do feel the same thing why can not we have the single layout pattern `Cards` and add the `cards -wthout-images and cards-with-images`

jeffchew commented:
> @jeffchew: yeah, I do feel the same thing why can not we have the single layout pattern `Cards` and add the `cards -without-images and cards-with-images`

@photodow I know you're still sorting out the reconciliation of these patterns on the BA side, but I'm wondering if it's worth at least merging this pattern with `CardsWithoutImages` for now and creating a variant for it? We need to think of a new name for it as well, which I think also is part of the efforts we identified today with pattern naming with @wonilsuhibm @ljcarot @larahanlon2

photodow commented:
> I'm wondering if it's worth at least merging this pattern with CardsWithoutImages for now and creating a variant for it?

I think this is the way to go. We could rename it to something like `CardSection` until we iron out the naming all of these. I don't know if it even needs to be a "variation" as it is in one we pass in images, and the other we don't. 🤷‍♂️

I haven't done the work to map all of this out yet, but I'm also seeing that this part could be extracted out to be reused in other places. I've seen many designs where this is used with other types of content.

![image](https://user-images.githubusercontent.com/3793636/70653104-73c0d980-1c19-11ea-9ae2-5a92409401cd.png)

sannminwin commented:
> @photodow @sannminwin @mkothur a general question, does Cards With Images need to be a completely separate layout pattern from Cards Without Images?

I completely agree that these two patterns should be merged.

mkothur commented:
PR updated with below:
- @jeffchew: moved the `matchHeight`  function inside the `packages/utilities` folder 
- @RaphaelAmadeu  addressed your comments. 
NOTE: @photodow addressing your comments on img element src and alt .

jeffchew commented:
> > I'm wondering if it's worth at least merging this pattern with CardsWithoutImages for now and creating a variant for it?
> 
> I think this is the way to go. We could rename it to something like `CardSection` until we iron out the naming all of these. I don't know if it even needs to be a "variation" as it is in one we pass in images, and the other we don't. 🤷‍♂️
> 
> I haven't done the work to map all of this out yet, but I'm also seeing that this part could be extracted out to be reused in other places. I've seen many designs where this is used with other types of content.
> 
> ![image](https://user-images.githubusercontent.com/3793636/70653104-73c0d980-1c19-11ea-9ae2-5a92409401cd.png)

Let's go with this approach then @mkothur

mkothur commented:
@jeffchew: So, I need to add `card with images` and `cards without images` into a single layout pattern called `CardsSection`?

mkothur commented:
@photodow, @jeffchew : Updated the PR by changing the layout pattern for cardSection. For now, I kept the other patterns `cardsWithImages` and `cardsWithoutImages` as it is. Based on your review will update these.

photodow commented on packages/styles/scss/patterns/cards-with-images/_cards-with-images-group.scss:
```diff
@@ -0,0 +1,125 @@
+  .bx--card-link {
+    background-color: $ui-01;
+    border-color: $ui-02;
+    outline: 1px solid $ui-background;
```
Also it looks like [L#47](https://github.com/carbon-design-system/ibm-dotcom-library/pull/803/files#diff-df575867f46cd4f3cb16dfdd18579e6cR47) contributes to this as well?

mkothur replied:
updated.

jeffchew commented on packages/patterns-react/src/internal/FeatureFlags.js:
```diff
@@ -77,3 +77,14 @@ export const DDS_CARDS_WITHOUT_IMAGES =
+ * @type {string | boolean}
+ */
+export const DDS_CARDS_WITH_IMAGES =
+  process.env.DDS_CARDS_WITH_IMAGES === 'true' || DDS_FLAGS_ALL || false;
```
Is `DDS_CARDS_WITH_IMAGES` needed anymore if we are consolidating to `DDS_CARDS_SECTION`?

mkothur replied:
yeah @jeffchew, we do not need, removed and updated.

jeffchew commented on packages/patterns-react/src/patterns/CardSection/CardSection.js:
```diff
@@ -0,0 +1,103 @@
+  };
+
+  return featureFlag(
+    DDS_CARDS_WITH_IMAGES,
```
switch to the new flag

mkothur replied:
yeah

jeffchew commented on packages/patterns-react/src/patterns/CardSection/CardsWithImages.js:
```diff
@@ -0,0 +1,89 @@
+        </div>
+      </div>
+    </section>
+  );
```
I'm a little confused about the structure now. Shouldn't it be `CardSection` > choose Card type > `card with image` or `card without image`? And are the "group" components redundant between the cards with or without images?

photodow commented on packages/patterns-react/src/patterns/CardSection/CardSection.js:
```diff
@@ -0,0 +1,103 @@
+      Component = CardsWithoutImagesGroup;
+    } else if (group.type === 'imageCards') {
+      Component = CardsWithImagesGroup;
+    }
```
I think to @jeffchew's point we've built this slightly backward.

###### CardSection

```jsx
function CardSection (cards) { 
  <h2>title</h2>
  cards.forEach((card) => {
    return card;
  })
}

export CardSection;
```
I did this quickly to hopefully demonstrate what we're striving to do, so ignore the imperfect syntax. `CardSection` is basically the pattern. You can either pass in an image, or don't pass in an image to the cards it's optional.

For the time being to avoid breaking anyone's code while we figure out naming and more, we can create wrappers that dictate the specific patterns originally created.

###### ImageCards
```jsx
import CardSection;

function ImageCards (data) {
  if images included in data
  <CardSection cards={images included} />
}

export ImageCards;
```

###### SimpleCards
```jsx
import CardSection;

function SimpleCards (data) {
  if no images in data
  <CardSection cards={no images} />
}

export SimpleCards;
```

`ImageCards`, and `SimpleCards` have the dependency on `CardSection`. Not the other way around.


Let me know if we need to jump on a call real quick to talk through this.

jeffchew replied:
@photodow do we anticipate `ImageCards` and `SimpleCards` to be two separate patterns that just use the underlying `CardSection` component, or we are introducing a `CardSection` pattern (the intent), with two variations to choose from?

photodow replied:
I think it kind of depends on how this pattern naming exercise pans out. No matter how you slice the dice the `CardSection` in my opinion has two variants. You can either include the images, or you don't.  These wrappers at the moment buys us some time with Ke and the fragility of Drupal at the moment. If the pattern naming work going on dictates a pattern for `ImageCards`, and `SimpleCards` then the outstanding question from me is will our adopters expect a pattern to be exported with that exact name. If not, then we deprecate these, update Ke and team about the changes. If they are required then using this wrapper method we're still sharing the same code, maybe add a little specific validation on the data, but we're not duplicating code this way at least we've mostly just added a pointer to the core `CardSection` with maybe an opinionated data model. Does that make sense?

jeffchew replied:
Got it, and agreed that regardless we need to build out `CardSection` no matter how it's sliced. I like the idea of passing `type` as a prop still, then the cards are rendered out accordingly. This also can open up the ability to add more card types in the future:

```
<CardSection type="image" data={{...}} />
```

If at the end of the day we need to create "decorator" pattern wrappers like `ImageCards` or `SimpleCards`, those can be a lightweight wrapper pattern that uses `CardSection` and explicitly setting the type, and also shaping the PropType shape accordingly based on the pattern type.

mkothur replied:
Okay, I will rewrite the components.

photodow replied:
> ```jsx
> <CardSection type="image" data={{...}} />

@jeffchew if we want to add the extra level of logic within the CardSection then we could totally go that route. I guess I question if it's necessary. What exactly would it do? Just validate the data and require images or not? The style, layout or function doesn't need to really change I don't think. The image and eyebrow would still need to be passed in via the data. CardLink should handle the optional eyebrow and image. I see it making the wrappers more simple assuming they are necessary in the first place. If it's not necessary and needs to be removed it adds a second touchpoint to remove.

jeffchew replied:
To your original point, I guess it all depends on how the naming things shakes out, but going with building out `CardSection` should get us through this for now and we can come back to it.

mkothur commented on packages/utilities/src/utilities/matchHeight/matchHeight.js:
```diff
@@ -0,0 +1,34 @@
+ * @private
+ * @param {*} parentEl Parent element selector
+ */
+const matchHeight = parentEl => {
```
Okay.

annawen1 commented on packages/patterns-react/src/patterns/CardSection/CardSection.js:
```diff
@@ -0,0 +1,103 @@
+      <div className={`${prefix}--cards-with-images__container`}>
+        <div className={`${prefix}--cards-with-images__row`}>
+          <div className={`${prefix}--cards-with-images__col`}>
+            {_renderCardsGroup(theme, cardsGroup, cardType)}
```
Don't see `cardType` being used in `_renderCardsGroup()`.

jeffchew commented:
@mkothur can you get rid of the "Cards Without Images" story in storybook if that's getting replaced by "SimpleCards"?

jeffchew commented on packages/patterns-react/.env.example:
```diff
@@ -4,6 +4,7 @@ PATTERNS_STORYBOOK_SOURCEMAPS=<Boolean to turn on/off sourcemaps in storybook>
 
 DDS_CARD_ARRAY=<Boolean flag to turn on/off the CardArray component>
+DDS_CARD_SECTION=<Boolean flag to turn on/off the CardSection component>
 DDS_CARDS_WITHOUT_IMAGES=<Boolean flag to turn on/off the CardsWithoutImages component>
```
Can get rid of this if we have moved to `CardSection`

mkothur replied:
Updated the PR.

mkothur commented:
Sure @jeffchew will do.

photodow commented on packages/styles/scss/patterns/card-section/_card-section.scss:
```diff
@@ -63,6 +65,7 @@
 
       &:hover {
         cursor: pointer;
+        background: $hover-ui;
```
Shouldn't this live in the theme mixin above?

photodow replied:
In general I'm seeing interaction issues with the cards themselves.

mkothur replied:
@photodow: you mean the hover class could be outside of mixin?

mkothur replied:
And interactions issues means like flickering while switching b/w cards?

photodow replied:
Low quality `gif`, but I'm not seeing any interaction. `:hover`, `:active`, `:focus` all seems off or none existent. It should just come in default via the `CardLink`, so I'm wondering if we overrode everything on accident.

![Jan-07-2020 13-45-16](https://user-images.githubusercontent.com/3793636/71924126-fd4dc380-3153-11ea-91ba-b2acfef64081.gif)

photodow commented on packages/styles/scss/patterns/card-section/_card-section.scss:
```diff
@@ -7,6 +7,8 @@
+  background: $ui-background;
   .bx--card-link {
     background-color: $ui-01;
     outline: 1px solid $ui-background;
```
I still don't think we should be using outline to do this. The condensed grid was designed to allow us to layout our cards in this way.

We're getting sone weird outlining issues.

<img src="https://user-images.githubusercontent.com/3793636/71921875-5cf5a000-314f-11ea-9d09-b8071b30c6a2.png" width="300px" />

mkothur replied:
oh I will remove that outline.

mkothur commented:
@photodow : yeah I have used same height utility function inside the cardSection and I did not see any broken changes.

mkothur commented:
@photodow : yeah I am using `object-fit: cover ` property in card link -image selector.

photodow commented on packages/styles/scss/components/card-link/index.scss:
```diff
@@ -17,12 +17,21 @@
+    &__image {
+      width: 100%;
+      height: auto;
+      object-fit: cover;
```
> @photodow : yeah I am using object-fit: cover property in card link -image selector.

Oh nice. That's what I get for trying to go quick on this. Sorry about that. I guess I thought a wrapper was required to make it work, but to my surprise that's not the case.

mkothur replied:
@photodow: I have addressed the above comments and updated the PR and let me know for ant changes.

photodow commented on packages/patterns-react/src/patterns/CardSection/CardSection.js:
```diff
@@ -0,0 +1,93 @@
+  const containerRef = useRef();
+
+  useEffect(() => {
+    root.addEventListener('resize', sameheight(containerRef.current, 'md'));
```
Here's what I'm seeing in ways of `sameheight`. Everything should align.

![image](https://user-images.githubusercontent.com/3793636/71923688-3174b480-3153-11ea-8156-988987d5db55.png)

It also looks like you're not calling it on `onload` unless I missed this too somewhere else. That's why I didn't see anything. On resize I'm sttill not even seeing any changes to the HTML to try and set them all to the sameheight.

mkothur replied:
I see, let me look into this.

mkothur replied:
I am calling the `sameheight` function inside the `useEffect` hook

mkothur replied:
@photodow: made some changes and it is working fine but the `sameheight` helper function has some bugs, had a short call with @RaphaelAmadeu  he is making changes in utility function.

jeffchew commented on packages/patterns-react/src/patterns/CardSection/index.js:
```diff
@@ -5,4 +5,4 @@
  */
 
-export { default as CardsWithoutImages } from './CardsWithoutImages';
+export { default as CardSection } from './CardSection';
```
@mkothur `SimpleCards` and `ImageCards` should probably be exported here too.

mkothur replied:
Okay

jeffchew commented on packages/patterns-react/src/patterns/CardSection/SimpleCards.js:
```diff
@@ -0,0 +1,24 @@
+  // });
+  return <CardSection {...props} />;
+};
+export default SimpleCards;
```
One thought. For `ImageCards` and `SimpleCards`, we probably should define the prop types for these, so that we ensure that the props getting passed in follow the corresponding card types.

mkothur replied:
oh yeah will do that.

mkothur commented:
@jeffchew: updated the PR with the above changes.

mkothur commented:
```NOTE: To review the SameHeight function please checkout the @RaphaelAmadeu latest PR of the same height responsive fix.```

annawen1 commented on packages/styles/scss/components/card-link/index.scss:
```diff
@@ -17,12 +17,21 @@
+    &__image {
+      width: 100%;
+      height: auto;
+      object-fit: cover;
```
images look a little small, I thought you had a height set to them? 240px? and looks like there's supposed to be a border between the cards

<img width="1078" alt="Screen Shot 2020-01-08 at 10 03 02 AM" src="https://user-images.githubusercontent.com/54281166/71988605-32165500-31fe-11ea-8ad5-8a6b11e49b33.png">

mkothur replied:
yeah @annawen1 as per @photodow I have removed the gray border on the cards. And I have changed the height to `auto` instead of having a specific height.

annawen1 replied:
I thought @photodow was referring to use of `outline`? Not sure if we wanted to remove the spacing/border from the designs though. @photodow?

annawen1 replied:
I'm fine with going ahead and merging it now since this PR has been open for a while. We can open a ticket to address the border.

annawen1 replied:
I'm fine with going ahead and merging it now since this PR has been open for a while. We can open a ticket to address the border.

photodow replied:
@mkothur naw, @annawen1's right. We need the spacing inbetween. That comes from the Condensed Grid. We don't have to build that into each component, we just have to use what's already there. If Carbon decides one day they want to change the condensed grid to 3px, then we have to go to a bunch of components and patterns to make sure we updated.

Here are a couple resources...
https://carbon-elements.netlify.com/grid/examples/preview/#condensed-grid
https://www.carbondesignsystem.com/guidelines/2x-grid/implementation/#the-condensed-grid

Reach out if you have any questions or concerns.

photodow replied:
As for the height, we're going to need a specific ratio for the images if I'm not mistaken.

jeffchew replied:
New issue created to track this: https://github.com/carbon-design-system/ibm-dotcom-library/issues/976

jeffchew commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -21,11 +21,3 @@
-<!-- *** "package: vanilla" -->
-<!-- *** "package: services" -->
-<!-- *** "package: utilities" -->
-<!-- *** "package: styles" -->
```
@mkothur this might have been a bad merge, looks like changes from a day or so ago are gone...
